x/nexus/keeper/rate_limit.go

```diff
@@ -0,0 +1,130 @@
+package keeper
+
+import (
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/axelar-core/utils"
+	"github.com/axelarnetwork/axelar-core/utils/events"
+	"github.com/axelarnetwork/axelar-core/utils/key"
+	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	"github.com/axelarnetwork/axelar-core/x/nexus/types"
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+// RateLimitTransfer applies a rate limit to transfers, and returns an error if the rate limit is exceeded
+func (k Keeper) RateLimitTransfer(ctx sdk.Context, chain exported.ChainName, asset sdk.Coin, outgoing bool) error {
```

Comment:
```suggestion
func (k Keeper) CheckRateLimit(ctx sdk.Context, chain exported.ChainName, asset sdk.Coin, outgoing bool) error {
```

Reply:
I didn't use that since we are not just checking the rate, but also incrementing the tracked amount and preventing the transfer if it exceeds it.